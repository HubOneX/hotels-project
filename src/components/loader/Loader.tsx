import { CircularProgress, Container } from "@mui/material";
import Typography from "@mui/material/Typography";
import React from "react";

const Loader = () => {
  return (
    <Container
      sx={{
        display: "flex",
        flexDirection: "column",
        alignItems: "center",
        marginTop: "40vh",
      }}
    >
      <Typography sx={{ marginBottom: "20px", fontSize: "20px" }}>
        Loading data...
      </Typography>
      <CircularProgress size={80} />
    </Container>
  );
};

export default Loader;
