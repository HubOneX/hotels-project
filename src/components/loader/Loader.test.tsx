import React from "react";
import renderer from "react-test-renderer";
import Loader from "./Loader";

describe("Loader", () => {
  it("should match the snapshot", () => {
    const container = renderer.create(<Loader />);

    expect(container).toMatchSnapshot();
  });
});
