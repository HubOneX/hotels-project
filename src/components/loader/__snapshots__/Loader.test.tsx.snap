// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match the snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-15jobu-MuiContainer-root"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1dtp7xy-MuiTypography-root"
  >
    Loading data...
  </p>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 80,
        "width": 80,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;
